src/logic/basic.lean

```diff
@@ -1112,6 +1112,15 @@ lemma apply_ite {α β : Type*} (f : α → β) (P : Prop) [decidable P] (x y :
   f (ite P x y) = ite P (f x) (f y) :=
 apply_dite f P (λ _, x) (λ _, y)
 
+lemma apply_dite2 {α β γ : Type*} (f : α → β → γ) (P : Prop) [decidable P] (a : P → α)
```

Comment:
Do we have other instances of this naming scheme? I would guess that the more long-winded `apply_dite_dite` is more common.

Reply:
Can both of these lemmas be `@[simp]` lemmas?

Reply:
We have the `image2` and `image3` functions for two-argument and three-argument functions, at least, which is probably the example I was following here.

Reply:
I've marked them both `@[simp]`. `apply_dite` and `apply_ite` aren't marked `@[simp]`, but there's a reasonable case that these new lemmas are genuinely simplifying by reducing the number of `dite` or `ite` uses, when the single-argument versions don't actually make things simpler.

Reply:
I've reverted that `@[simp]` marking, as it turns out it fails linting. (This file is too early in the import hierarchy to be able to test that with `#lint` before commit.)

Reply:
Ah, I see, it fails:

```
/- The `simp_var_head` linter reports: -/
/- LEFT-HAND SIDE HAS VARIABLE AS HEAD SYMBOL.
Some simp lemmas have a variable as head symbol of the left-hand side: -/
-- logic/basic.lean
#print apply_dite2 /- Left-hand side has variable as head symbol: f -/
#print apply_ite2 /- Left-hand side has variable as head symbol: f -/
```

Sorry, I should have expected that.